### DISCUSSION TITLE
companion app isn't detected

### DISCUSSION THREAD

**Discussion by rastoboy** [aclap-dev/video-downloadhelper#835] [Help and Q&A]
### Your question

Hi, I can't seem to get the companion app detected.  I've answered the Firefox popup for permission affirmatively.  It just no go!

~/.local/share/vdhcoapp/vdhcoapp --info
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/rasto/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/rasto",
  "converterBinary": "/home/rasto/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

uname -a
Linux morgoth 6.5.0-35-lowlatency #35.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May  6 16:41:14 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Firefox version: 126.0.1

Any input would be greatly appreciated!

### Addon & CoApp version and details

~/.local/share/vdhcoapp/vdhcoapp --info
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/rasto/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/rasto",
  "converterBinary": "/home/rasto/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

uname -a
Linux morgoth 6.5.0-35-lowlatency #35.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May  6 16:41:14 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Firefox version: 126.0.1

Any input would be greatly appreciated!

**Comment by Mattmjs** [comment]
How did you install the companion app. If using gui in Ubuntu maybe it wasn't properly registered. Which requires a terminal 
command to fully register the companion app. Have you referred to the wiki: 
https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#linux-all

**Comment by rastoboy** [comment]
Thanks for your response.  Yes, I've gone through that page.  I installed via the commandline, and it said it was registered.  

Does it matter if it's a youtube video I'm trying to download?  I'm only installing in FF because Google told me they despise me as a user and won't let it work in Chrome with youtube lol.

**Reply by Mattmjs** [reply]
OK that is puzzling if it said it was registered. On the same wiki page there are some commands to run diagnostics.
You should try that and share what it says here.

**Reply by GrampaWildWilly** [reply]
Oh they despise everybody.  It is a legal restriction that Google can impose because they are a monopoly & own both the Chrome browser & the YouTube web site.  The restriction is that no Chrome browser extension available on the Chrome Web Store is permitted to download from YouTube.  I imagine they have their own browser extension that does that.  I wouldn't know.  I've never used Chrome.  From everything I've been able to glean second hand reading stuff in this forum & elsewhere, Firefox is superior to Chrome anyway.  As for your particular problem, I can't help.  I don't know Linux any more than I know Chrome.  Once you get it installed & if you happen to hit a problem downloading something, I might be able to help with that.  But I'm just another user & I don't have the expertise you appear to need here.

**Reply by GrampaWildWilly** [reply]
There's been some posts in here from Linux users saying they had to duplicate a certain file from one directory to another to make things work.  I think it was a json of some name.  Again, I know exactly diddly squat about Linux so I only barely grasped what they were saying.  You might search this forum for those discussions & see if they are any help.

**Reply by GrampaWildWilly** [reply]
Oh.  If you're just now installing Firefox for the first time, you'd best install the latest beta of VDH.  No sense in falling into bugs that have already been solved.  Make sure any bugs you hit are totally brand spanking new.
